src/System.Drawing.Common/src/System/Drawing/Region.cs

```diff
@@ -198,7 +198,18 @@ public void ReleaseHrgn(IntPtr regionHandle)
                 throw new ArgumentNullException(nameof(regionHandle));
             }
 
-            SafeNativeMethods.IntDeleteObject(new HandleRef(this, regionHandle));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I think I would prefer to have this logic in partial files for Windows and Unix, rather than do checks at runtime.